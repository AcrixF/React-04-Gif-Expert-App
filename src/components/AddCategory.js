import React, { useState } from 'react';

const AddCategory = () => {

    const [inputValue, setInputValue] = useState('Brenda Marlen')

    const handleInputChange = (event) => {
        setInputValue(event.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log(e);
    }

    return(
        <form onSubmit={handleSubmit}>
            <input 
                type="text" 
                value = {inputValue}
                onChange = { handleInputChange }/>
        </form>
    );
};

export default  AddCategory;