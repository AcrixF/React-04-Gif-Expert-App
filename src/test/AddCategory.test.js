import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import AddCategory from '../components/AddCategory';

describe('should show the <AddCategory/> component', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories ={ setCategories } />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories ={ setCategories } />);
    })    


    test('should show the component snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should simulate change the input', () => {
        const value = 'Hunter x Hunter';
        const input = wrapper.find('input');
        input.simulate('change', { target: value });
    });

    test('should not submit the data', () => {
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        expect(setCategories).not.toHaveBeenCalled();
    });
        
});
