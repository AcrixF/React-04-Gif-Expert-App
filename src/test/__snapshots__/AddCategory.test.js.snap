// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show the <AddCategory/> component should show the component snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
